File: api/BeSureApi/Models/VendorBranch.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace BeSureApi.Models
{
    public class VendorBranch
    {
        [Required(ErrorMessage = "validation_error_vendorbranchbranch_create_tenantoffice_required")]
        public int TenantOfficeId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_code_required")]
        public string Code { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_name_required")]
        public string Name { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_address_required")]
        public string Address { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_vendorid_required")]
        public int VendorId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_city_required")]
        public int CityId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_state_required")]
        public int StateId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_country_required")]
        public int CountryId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_pincode_required")]
        public string Pincode { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_contactname_required")]
        public string ContactName { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_email_required")]
        public string Email { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_primaryccode_required")]
        public string ContactNumberOneCountryCode { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_primarycname_required")]
        public string ContactNumberOne { get; set; }
        public string? ContactNumberTwoCountryCode { get; set; }
        public string? ContactNumberTwo { get; set; }
        public string? TollfreeNumber { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_creditperiod_indays_required")]
        public string CreditPeriodInDays { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_gstno_required")]
        public string GstNumber { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_gstvendortype_required")]
        public int GstVendorTypeId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_gstarn_required")]
        public string GstArn { get; set; }
        public string? Remarks { get; set; }
    }
    public class VendorBranchUpdate
    {
        public int Id { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_tenantoffice_required")]
        public int TenantOfficeId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_code_required")]
        public string Code { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_name_required")]
        public string Name { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_address_required")]
        public string Address { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_city_required")]
        public int CityId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_state_required")]
        public int StateId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_country_required")]
        public int CountryId { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_pincode_required")]
        public string Pincode { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_contactname_required")]
        public string ContactName { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_email_required")]
        public string Email { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_primaryccode_required")]
        public string ContactNumberOneCountryCode { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_primarycname_required")]
        public string ContactNumberOne { get; set; }
        public string? ContactNumberTwoCountryCode { get; set; }
        public string? ContactNumberTwo { get; set; }
        public string? TollfreeNumber { get; set; }
        [Required(ErrorMessage = "validation_error_vendor_create_creditperiod_indays_required")]
        public int CreditPeriodInDays { get; set; }
        [Required(ErrorMessage = "validation_error_vendor_create_gstno_required")]
        public string GstNumber { get; set; }
        [Required(ErrorMessage = "validation_error_vendorbranch_create_gstvendortype_required")]
        public int GstVendorTypeId { get; set; }
        public string GstArn { get; set; }
        public string? Remarks { get; set; }
        public bool IsActive { get; set; }
    }

    public class VendorBranchList
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public int VendorId { get; set; }
        public string Name { get; set; }
        public string City { get; set; }
        public string ContactName { get; set; }
        public string ContactNumberOneCountryCode { get; set; }
        public string ContactNumberOne { get; set; }
        public string TenantLocation { get; set; }
        public bool IsActive { get; set; }
    }
}

File: ui/src/types/vendorBranch.ts
import { Decoder, array, boolean, nullable, number, object, string } from "decoders";

export interface VendorBranchCreation {
  Name: string
  Address: string
  TenantOfficeId: number
  CityId: number
  VendorId: number
  StateId: number
  Pincode: string
  ContactName: string
  Email: string
  ContactNumberOneCountryCode: string
  ContactNumberOne: string
  ContactNumberTwoCountryCode: string
  ContactNumberTwo: string
  CreditPeriodInDays: string
  GstNumber: string
  GstVendorTypeId: number
  CountryId: number
  GstArn: string
  Remarks: string
  TollfreeNumber: string
  Code: string
}

export interface Select {
  value: any,
  label: any,
  code?:any
}

export interface SelectDetails {
  Select: Select[]
}

export interface VendorBranchSelectDetails {
  States: Select[],
  Cities: Select[],
  Countrys: Select[],
  Location: Select[],
  GstVendorType: Select[],
  PrimaryCountryCode:Select[]
}

export interface VendorSearchForFilter {
  SearchText: string;
}

export interface VendorBranchList {
  Id: number;
  Code: string;
  Name: string;
  City: string;
  ContactNumberOne: string;
  ContactNumberOneCountryCode: string;
  ContactName: string;
  TenantLocation: string;
  IsActive: boolean;
}

export const vendorBranchDetailsDecoder: Decoder<VendorBranchList> = object({
  Id: number,
  Code: string,
  VendorId: number,
  Name: string,
  City: string,
  ContactNumberOneCountryCode: string,
  ContactNumberOne: string,
  ContactName: string,
  TenantLocation: string,
  IsActive: boolean,
});

export interface MultipleVendorBranchDetails {
  VendorBranches: VendorBranchList[];
  TotalRows: number;
  PerPage: number;
}

export const multipleVendorBranchDetailsDecoder: Decoder<MultipleVendorBranchDetails> = object({
  VendorBranches: array(vendorBranchDetailsDecoder),
  TotalRows: number,
  PerPage: number
});

export interface VendorBranchCreateResponse {
  IsVendorBranchCreated: Boolean;
}

export const vendorBranchCreateResponseDecoder: Decoder<VendorBranchCreateResponse> = object({
  IsVendorBranchCreated: boolean
})

export interface VendorBranchEdit {
  Id: number,
  Name: string
  Address: string
  TenantOfficeId: number
  CityId: number
  StateId: number
  Pincode: string
  ContactName: string
  Email: string
  ContactNumberOneCountryCode: string
  ContactNumberOne: string
  ContactNumberTwoCountryCode: string | null
  ContactNumberTwo: string | null
  CreditPeriodInDays: number
  GstNumber: string
  GstVendorTypeId: number
  CountryId: number
  GstArn: string
  Remarks: string | null
  TollfreeNumber: string | null
  Code: string
  IsActive: boolean
}

export const vendorBranchEditDecoder: Decoder<VendorBranchEdit> = object({
  Id: number,
  Name: string,
  Address: string,
  TenantOfficeId: number,
  CityId: number,
  StateId: number,
  Pincode: string,
  ContactName: string,
  Email: string,
  IsActive: boolean,
  ContactNumberOneCountryCode: string,
  ContactNumberOne: string,
  ContactNumberTwoCountryCode: nullable(string),
  ContactNumberTwo: nullable(string),
  CreditPeriodInDays: number,
  GstNumber: string,
  GstVendorTypeId: number,
  CountryId: number,
  GstArn: string,
  Remarks: nullable(string),
  TollfreeNumber: nullable(string),
  Code: string,
})

export interface VendorBranchEditDetails {
  VendorBranchDetails: VendorBranchEdit;
}

export const vendorBranchEditDetailsDecoder: Decoder<VendorBranchEditDetails> = object({
  VendorBranchDetails: vendorBranchEditDecoder
})

export interface VendorBranchEditResponse {
  IsVendorBranchUpdated: Boolean;
}

export const vendorBranchEditResponseDecoder: Decoder<VendorBranchEditResponse> = object({
  IsVendorBranchUpdated: boolean
})

export interface VendorBranchDeleteResponse {
  IsDeleted: Boolean;
}

export const vendorBranchDeleteResponseDecoder: Decoder<VendorBranchDeleteResponse> = object({
  IsDeleted: boolean
})

export interface BranchInVendorDetail {
  Id: number;
  BranchName: string;
  Address:string;
  GstStateCode:string;
}
export const  branchInVendorDetailDecoder: Decoder<BranchInVendorDetail> = object({
  Id: number,
  BranchName: string,
  Address:string,
  GstStateCode:string
});

export interface BranchInVendorList {
  VendorBranches: BranchInVendorDetail[];
}

export const branchInVendorListDecoder: Decoder<BranchInVendorList> = object({
  VendorBranches: array( branchInVendorDetailDecoder),
});
